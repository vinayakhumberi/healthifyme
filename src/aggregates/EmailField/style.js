import styled from 'styled-components';
import { rem } from '../../utils/style.util';

export const Container = styled.div`
`;

export const MessageSpace = styled.div`
  height: ${rem(21)};
  display: block;
`;