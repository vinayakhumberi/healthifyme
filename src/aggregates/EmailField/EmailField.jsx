import React, {useState, useEffect, useRef} from "react";
import PropTypes from "prop-types";
import { Container, MessageSpace } from "./style";
import Text from '../../components/Text/Text';
import TextInput from '../../components/TextInput/TextInput';
import {
  EMAIL_REGEX
} from '../../constants/general.constants';

const EmailField = function (props) {
  const [showMessage, setShowMessage] = useState('');
  const [valid, setValid] = useState(false);
  const emailRef = useRef('');
  useEffect(() => {
    props.updateEmail({
      value: emailRef.current,
      status: valid
    })
  }, [valid]);
  const handleTextChange = (e) => {
    const data = e.target.value;
    if (showMessage) setShowMessage("");
    if (valid) setValid(false);
    emailRef.current = data;
    verifyEmail(data, false);
  };
  const handleVerify = (e) => {
    const data = e.target.value;
    verifyEmail(data, true);
  };

  const verifyEmail = (data, error) => {
    if (data.length === 0 && error) {
      setShowMessage("Email is mandatory");
    } else if (data.length > 0 && data.length <= 5 && error) {
      setShowMessage("Should me more than 5 characters long");
    } else {
      if (EMAIL_REGEX.test(data)) {
        setValid(true);
      } else {
        if (error) setShowMessage("Should be a valid email");
      }
    }
  };
  return (
    <Container>
      <TextInput
        elementType="text"
        marginTop={10}
        marginBottom={0}
        paddingTop={10}
        paddingBottom={10}
        paddingLeft={10}
        paddingRight={10}
        width="100%"
        placeHolder="Enter your Email"
        borderColor={showMessage ? '#ec445c' : '#eee'}
        onChange={handleTextChange}
        onBlur={handleVerify}
      />
      <MessageSpace>
        {showMessage && <Text
          text={showMessage}
          color="#ec445c"
          paddingTop={5}
          paddingLeft={5}
        />}
      </MessageSpace>
    </Container>
  );
};
EmailField.propTypes = {
  updateEmail: PropTypes.func
};

EmailField.defaultProps = {
  updateEmail: () => {}
};

export default EmailField;
