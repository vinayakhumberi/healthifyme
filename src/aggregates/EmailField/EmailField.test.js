import React from 'react';
import { mount, shallow, render } from 'enzyme';
import EmailField from './EmailField';

describe('<EmailField />', () => {
  let wrapper;
  const updateEmail = () => {};
  beforeEach(() => wrapper = shallow(<EmailField updateEmail={updateEmail} />));

  it('Should Render a EmailField Container', () => {
    expect(wrapper.find('[data-test="email-field-container"]').length).toEqual(1);
  });
  // it('Should pass updateEmail prop to TextInput', () => {
  //   console.log(wrapper.prop('updateEmail'));
  //   expect(wrapper.find('[data-test="email-field-container"]').get(0).props.updateEmail).toBeDefined();
  // });
});


describe('<EmailField /> error check', () => {
  let wrapper;
  const updateEmail = () => { };
  beforeEach(() => wrapper = mount(<EmailField updateEmail={updateEmail} />));
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  // it('Should Render a EmailField Container', () => {
  //   expect(wrapper.find('[data-test="email-field-container"]').length).toEqual(1);
  // });
  it('Should pass updateEmail prop to TextInput', () => {
    const input = wrapper.find('input');
    input.simulate('focus');
    input.simulate('change', { target: { value: 'Test Data' } });
    // expect(input.get(0).value).toEqual('Test Data');
  });
  it('Should show Error message is text is not more than 5 character long', () => {
    
  });
  it('Should show Error message if onBlur is fired and text is empty', () => {

  });
  it('Should show Error message if it is an invalid Emails', () => {

  });
});
