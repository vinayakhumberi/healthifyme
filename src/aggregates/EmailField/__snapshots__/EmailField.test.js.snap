// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailField /> error check renders correctly 1`] = `ReactWrapper {}`;
